File: SEM 3/OOPL/LAB2/q3.java
import java.util.Scanner;

class q3
{
    public static void main(String args[])
    {
        int i,n,ele,pos=-1;
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter Number of Elements : ");
        n=sc.nextInt();
        int a[]=new int[n];
        for(i=0;i<n;i++)
        {
            a[i]=sc.nextInt();
        }
        System.out.println("Enter element to search : ");
        ele=sc.nextInt();
        for(i=0;i<n;i++)
        {
            if(a[i]==ele)
            {
                pos=i;
            }
        }
        if(pos==-1)
        {
            System.out.println("Element not found");
        }
        else
        {
            System.out.println("Element found at position "+(pos+1));
        }
    }
}

File: SEM 6/ITL/Week_3/Lab3/Q3.py
def binarySearch(ll,l,r,ele):
	if l<=r:
		m = l + (r-l)//2
		if ll[m]==ele:
			return True,m
		elif ll[m] > ele:
			return binarySearch(ll,l,m-1,ele)
		else:
			return binarySearch(ll,m+1,r,ele)
	else:
		return False,-1

def main():
	ll = [3,5,6,2,1,7,8,9]
	ll.sort()
	num = int(input("Enter Number to Search : "))
	flag,pos = binarySearch(ll,0,len(ll)-1,num)
	if flag:
		print("Number Found at Position ",pos+1)
	else:
		print("Number not found")

if __name__ == "__main__":
	main()
